src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -612,6 +770,101 @@ public override void Dispatch(Cci.MetadataVisitor visitor)
         public override Cci.INestedTypeReference AsNestedTypeReference => this;
     }
 
+    /// <summary>
+    /// A type synthesized for each eligible string literal to hold the lazily-initialized string.
+    ///
+    /// https://github.com/dotnet/roslyn/blob/main/docs/features/string-literals-data-section.md
+    /// </summary>
+    internal sealed class DataSectionStringType : DefaultTypeDef, Cci.INestedTypeDefinition
```

Comment:
>DefaultTypeDef [](http://example.com/codeflow?start=50&length=14)

I think it would be good to extract a common abstract base "NestedTypeDefinition" to share across ```ExplicitSizeStruct``` and ```DataSectionStringType```